db_stress_tool/expected_value.cc

```diff
@@ -10,37 +10,31 @@
 #include <atomic>
 
 namespace ROCKSDB_NAMESPACE {
-bool ExpectedValue::Put(bool pending) {
-  if (pending && (PendingWrite() || PendingDelete())) {
-    return false;
-  }
-
+void ExpectedValue::Put(bool pending) {
   if (pending) {
     SetPendingWrite();
   } else {
     SetValueBase(NextValueBase());
     ClearDeleted();
     ClearPendingWrite();
   }
-  return true;
 }
 
-bool ExpectedValue::Delete(bool pending) {
-  if (pending && (PendingWrite() || PendingDelete())) {
-    return false;
-  }
+void ExpectedValue::Delete(bool pending, bool* existed) {
+  assert(existed);
 
   if (!Exists()) {
-    return false;
+    *existed = false;
```

Comment:
nit: can just return a bool instead of using bool*, which is what we do before #12838.

Reply:
fixed